packages/web/src/features/translation/TranslationSidebar.tsx

```diff
@@ -294,12 +307,16 @@ export const TranslationSidebar = forwardRef<
                         <RichTextInput
                           ref={translatorNotesEditorRef}
                           name="translatorNoteContent"
-                          value={translatorNoteContent}
                           onBlur={() => saveTranslatorNote.flush()}
                           onChange={saveTranslatorNote}
```

Comment:
Using refs like we are is a code smell. The key problem is that when the rich text content changes, we aren't updating `translatorNoteContent` and `footnoteContent`. Here is why that is a problem:

1. Find a verse with no notes. In this case whenever you switch words, `footnoteContent` is set to the empty string.
2. Type a footnote. The rich text input's state as well as the server state have the content of the note, but `footnoteContent` does not.
3. Switch words to one without a note. Now the sidebar will reset `footnoteContent` to the contents of the note for that word which is the same empty string. If it had changed, then the rich text input would reset with the new content, but in this case it didn't because the value being passed didn't change

If we change this onChange handler to something like the following, the issue is resolved:
```ts
onChange={content => {
  setTranslatorNoteContent(content)
  saveTranslatorNote(content)
})
```
Now whenever you type in the rich text input, the content states will change in the sidebar so that when they are reset when the word changes, the rich text input actually notices that the content changes and it needs to reset

Reply:
Thanks for the detailed explanation!

Reply:
Done.

Reply:
Alright, done now.